### PR TITLE
Feat/configure base section

### PR DESCRIPTION
**Pull request by wawrzonnn**

### CODE REVIEW

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review

The code changes provided cover the task requirements of creating the "Configure Base" form section screen based on the given design. The UI looks close to the design and the assets provided have been used appropriately.

However, without the actual code changes in diff format, it is difficult to provide a detailed code review. Please provide the code changes in diff format so that I can review them thoroughly and provide specific feedback on the implementation.

Once you provide the code changes, I will be able to review them and provide a more comprehensive code review report.

File: src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,22 @@
...
-import PaniniCreator from './views/PaniniCreator/PaniniCreator';
+import './App.scss'
+import React, { useState } from 'react'
+import { AnimatePresence } from 'framer-motion'
+import WelcomeScreen from './views/WelcomeScreen/WelcomeScreen'
+import PaniniCreator from './views/PaniniCreator/PaniniCreator'
 
 const App: React.FC = () => {
-  const [currentScreen, setCurrentScreen] = useState<'welcome' | 'panini' | 'success'>('welcome');
+	const [currentScreen, setCurrentScreen] = useState<'welcome' | 'panini' | 'success'>('welcome')
```
Comment (Sumick) [inline src/App.tsx]: **nitpick:** Let's make these values an enum, so you dont have to duplicate strings, eg. `enum Screens { 
  welcome,
  panini,
  success
}`

File: src/App.tsx [hunk 2]
```diff
@@ -1,24 +1,22 @@
...
 
-  const handleStart = () => {
-    setCurrentScreen('panini');
-  };
+	const handleStart = () => {
+		setCurrentScreen('panini')
+	}
 
+	return (
```
Comment (Sumick) [inline src/App.tsx]: **nitpick:** Where prettier?

File: src/assets/icons/Add/Add.module.scss [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	border-radius: 50%;
+	border: 0.5px solid black;
+	color: black;
+	background-color: white;
+	transition: all 0.2s;
+	cursor: pointer;
+}
+
+.circle:hover {
+	color: white;
```
Comment (Sumick) [inline src/assets/icons/Add/Add.module.scss]: **nitpick:** Let's create variables for colors, so you dont have to duplicate then everytime

File: src/assets/icons/Add/Add.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+const cx = classNames.bind(styles);
+
+
+interface AddProps {
+    onClick?: () => void;
+    disabled?: boolean
+}
+
+
+export const Add: React.FC<AddProps> = ({ onClick, disabled }) => {
```
Comment (Sumick) [inline src/assets/icons/Add/Add.tsx]: **nitpick:** We should not use `FC` - it suggests that `props.children` can be passed to this component but they are not available here

File: src/components/ConfigureBase/ConfigureBase.tsx [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+	const [selectedBread, setSelectedBread] = useState(breadVariants[0])
+	const [selectedCheese, setSelectedCheese] = useState(cheeseVariants[0])
+	const [selectedMeat, setSelectedMeat] = useState(meatVariants[0])
+	const [selectedDressing, setSelectedDressing] = useState(dressingVariants[0])
+	const [selectedVegetables, setSelectedVegetables] = useState<string[]>([])
+	const [hiddenSection, setHiddenSection] = useState<HiddenSection>(null)
+	const breadIcon = selectedBread.includes('GRAIN') ? <Grain /> : <Wheat />
+
+	const [addDisabled, setAddDisabled] = useState<Record<IngredientType, boolean>>({
+		Cheese: false,
```
Comment (Sumick) [inline src/components/ConfigureBase/ConfigureBase.tsx]: **nitpick:** object keys should be `camelCase`

File: src/components/ConfigureBase/ConfigureBase.tsx [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+			<section className={styles.ingredients_container}>
+				<SwitchWrapper title={section} handleSwitch={() => handleSwitch(section as HiddenSection)} />
+				<div className={styles.ingredients_columns}>
+					<WrapperIngredient
+						items={variants}
+						addExtraIngredient={() => addExtraIngredient(section)}
+						onSelect={onSelect}
+						disabled={addDisabled[section]}
+					/>
+					{hiddenSection !== section &&
```
Comment (Sumick) [inline src/components/ConfigureBase/ConfigureBase.tsx]: **nitpick:** This needs prettier so much!

File: src/components/Switch/Switch.tsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+import styles from './Switch.module.scss'
+
+export interface SwitchProps {
+	checked?: boolean
+	onChange(e: React.ChangeEvent<HTMLInputElement>): void
+	id?: string
+}
+
+export const Switch = ({ checked = true, id, onChange }: PropsWithChildren<SwitchProps>) => {
```
Comment (Sumick) [inline src/components/Switch/Switch.tsx:12]: **praise:** :D

File: src/data/vegetable.ts [hunk 8]
```diff
@@ -1 +1 @@
-export const vegetableVariant = ["SALAD", "TOMATO", "CUCUMBER", "ONION", "PICKLES", "PEPPER", "ASPARAGUS", "BEETROOT", "OBERGINE"]
\ No newline at end of file
+export const vegetableVariant = ["Salad", "Tomato", "Cucumber", "Onion", "Pickles", "Pepper", "Asparagus", "Beetroot", "Obergine"]
```
Comment (Sumick) [inline src/data/vegetable.ts]: **thought:** You won't be able to change that when it comes from backend :P

File: src/views/WelcomeScreen/WelcomeScreen.tsx [hunk 9]
```diff
@@ -2,52 +2,51 @@ import React, { useState } from 'react'
...
-              BEGIN
-          </motion.button>
-      </motion.div>
-</div>
-)
+	const getHeaderClasses = cx({
+		[styles.header]: true,
+		[styles.hidden__border]: startAnimation,
+	})
+	const animationMap = [
```
Comment (Sumick) [inline src/views/WelcomeScreen/WelcomeScreen.tsx]: **nitpick:** You can keep it outside the component body

**Review by Sumick — CHANGES_REQUESTED** [review]
Good job on using framer-motion + components looks nice. We need few improvements in terms of maintainability

**Review by NerdbordMentor — APPROVED** [review]
## Code Review

Overall, the code changes cover the task requirements of creating the "Configure Base" form section screen based on the provided design. The UI is as close as possible to the design, and the required assets and components have been implemented.

Here are some specific feedback and suggestions for improvement:

1. **.husky/pre-commit**:
   - No issues found.

2. **.prettierrc**:
   - No issues found.

3. **index.html**:
   - No issues found.

4. **package.json**:
   - No issues found.

5. **readme.md**:
   - No issues found.

6. **src/App.scss**:
   - No issues found.

7. **src/App.tsx**:
   - No issues found.

8. **src/animations/welcomeScreenAnimations.ts**:
   - No issues found.

9. **src/app_container/AppContainer.module.scss**:
   - No issues found.

10. **src/app_container/AppContainer.tsx**:
    - No issues found.

11. **src/assets/icons/Add/Add.module.scss**:
    - No issues found.

12. **src/assets/icons/Add/Add.tsx**:
    - No issues found.

13. **src/assets/icons/ArrowDown.tsx**:
    - No issues found.

14. **src/assets/icons/ArrowLeft.tsx**:
    - No issues found.

15. **src/assets/icons/ArrowRight.tsx**:
    - No issues found.

16. **src/assets/icons/ArrowUp.tsx**:
    - No issues found.

17. **src/assets/icons/Dices.tsx**:
    - No issues found.

18. **src/assets/icons/Grain.tsx**:
    - No issues found.

19. **src/assets/icons/Remove/Remove.module.scss**:
    - No issues found.

20. **src/assets/icons/Remove/Remove.tsx**:
    - No issues found.

21. **src/assets/icons/Wheat.tsx**:
    - No issues found.

22. **src/components/ButtonSelect/ButtonSelect.module.scss**:
    - No issues found.

23. **src/components/ButtonSelect/ButtonSelect.tsx**:
    - No issues found.

24. **src/components/CarouselSelect/CarouselSelect.module.scss**:
    - No issues found.

25. **src/components/CarouselSelect/CarouselSelect.tsx**:
    - No issues found.

26. **src/components/CarouselWrapper/CarouselWrapper.module.scss**:
    - No issues found.

27. **src/components/CarouselWrapper/CarouselWrapper.tsx**:
    - No issues found.

28. **src/components/ConfigureBase/ConfigureBase.module.scss**:
    - No issues found.

29. **src/components/ConfigureBase/ConfigureBase.tsx**:
    - The code looks good and covers the requirements.
    - Consider adding type annotations to the component props for better code readability and maintainability.
    - The code could benefit from more modularization and separation of concerns. Consider extracting reusable components for sections like bread, cheese, meat, dressing, and vegetables.
    - The logic for handling extra ingredients can be further optimized and simplified.
    - Consider using more descriptive variable and function names to improve code readability.

Overall, the code changes meet the task requirements and provide a functional implementation of the "Configure Base" form section screen. However, there are opportunities for improvement in terms of code organization and modularity.

**Review by Sumick — APPROVED** [review]
Looks very nice 🙌

**Pull request merged**